flexmeasures_client.py

```diff
@@ -80,14 +89,68 @@ def initialize(self):
         self.log(f"Car_max_soc: {self.CAR_MAX_SOC_IN_KWH} kWh.")
 
         if c.OPTIMISATION_MODE == "price":
-            self.FM_OPTIMISATION_CONTEXT = {"consumption-price-sensor": c.FM_PRICE_CONSUMPTION_SENSOR_ID, "production-price-sensor": c.FM_PRICE_PRODUCTION_SENSOR_ID}
+            self.FM_OPTIMISATION_CONTEXT = {"consumption-price-sensor": c.FM_PRICE_CONSUMPTION_SENSOR_ID,
+                                            "production-price-sensor": c.FM_PRICE_PRODUCTION_SENSOR_ID}
         else:
             # Assumed optimisation = emissions
-            self.FM_OPTIMISATION_CONTEXT = {"consumption-price-sensor": c.FM_EMISSIONS_SENSOR_ID, "production-price-sensor": c.FM_EMISSIONS_SENSOR_ID}
+            self.FM_OPTIMISATION_CONTEXT = {"consumption-price-sensor": c.FM_EMISSIONS_SENSOR_ID,
+                                            "production-price-sensor": c.FM_EMISSIONS_SENSOR_ID}
         self.log(f"Optimisation context: {self.FM_OPTIMISATION_CONTEXT}")
 
+        # Initially assume no problems untill attempts for retrieving schedules have failed.
+        self.set_state("input_boolean.error_schedule_cannot_be_retrieved", state="off")
+
+        # Ping every half hour. If offline a separate process will run to increase polling frequency.
+        self.connection_error_counter = 0
+        self.run_every(self.ping_server, "now", 30 * 60)
+        self.set_state("input_boolean.is_flexmeasures_live", state="on")
+        self.handle_for_repeater = ""
+        # For testing ping
+        # self.test_counter = 0
+
         self.log("Completed initializing FlexMeasuresClient")
 
+    def ping_server(self, *args):
+        """ Ping function to check if server is alive """
+        url = c.FM_PING_URL
+        # Test code:
+        # self.test_counter += 1
+        # if self.test_counter in range(2, 9):
+        #     url = url + "fout"
```

Comment:
@nhoening There's your explanation to your mystery log statement.

Reply:
Yep, will remove in next update. Should not have been in there.